kvdb/etcd/config.go

```diff
@@ -25,4 +25,6 @@ type Config struct {
 	InsecureSkipVerify bool `long:"insecure_skip_verify" description:"Whether we intend to skip TLS verification"`
 
 	CollectStats bool `long:"collect_stats" description:"Whether to collect etcd commit stats."`
+
+	ReadWriteLock bool `long:"rw_lock" description:"Ensure single writer by RW locking all transactions."`
```

Comment:
After #5547, fully remote mode is no longer a mode but always on. Setting `ReadWriteLock` to false is unsafe with a fully remote database, so is there a use for this switch?

Reply:
Yeah, maybe it would be less confusing to users if this were an argument passed to `kvdb.Open()/Create()` instead of an option in the config. I assume you intended this to be used like the namespace (e.g. `cfg.CloneWithReadWriteLock()`). But I agree that it being exposed as a config option to the user probably is confusing.

Reply:
We can remove config option and add it as a param to `Open/Create`.  Really no need for it to be user exposed, since the idea is that this is only used internally until we make sure all legacy code is safe to use concurrently.